### ISSUE TITLE
OnlyRule not work with custom_rules

### ISSUE THREAD

**Issue by Ernest0-Production** [realm/SwiftLint#6029]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

If you specify a custom rule in the `--only-rule` argument, then no violations are detected during linting. (It does scan the files though)

When linting a built-in rule, everything works correctly

```bash
$ swiftlint lint --no-cache --only-rule my_rule
```

### Environment

* SwiftLint version 0.58.2
* Xcode version 16.2
* Installation method used mise-en-place

**Comment by SimplyDanny** [comment]
Historically, handling of custom rules has its peculiarities and we have one of these examples here. Unexpectedly, the command to run a single custom rule is like `swiftlint lint --only-rule custom_rules --only-rule my_rule`.
